pkg/ds/daemon_set.go

```diff
@@ -45,6 +48,8 @@ type DaemonSet interface {
 	// Returns a list of all nodes that are selected by this daemon set's selector
 	EligibleNodes() ([]types.NodeName, error)
 
+	metricName(name string) string
```

Comment:
warning: unexported function in interface prevents anyone outside this package from implementing the interface.

$GOPATH/pkgb/hi.go

```go
package pkgb

// Greeter can greet things.
type Greeter interface {
  greet(string) string
}
```

$GOPATH/pkga/hi.go

```go
package main

import "fmt"
import "pkgb"

type helloer struct{}

func (h *helloer) greet(name string) string {
  return "Hello, " + name + "!"
}

var _ pkgb.Greeter = &helloer{}

func main() {
  h := &helloer{}
  fmt.Println(h.greet("World"))
}
```

build output:

```
./hi.go:12: cannot use helloer literal (type *helloer) as type pkgb.Greeter in assignment:
	*helloer does not implement pkgb.Greeter (missing pkgb.greet method)
		have greet(string) string
		want pkgb.greet(string) string
```

I don't know if we care enough to change that.